clients/unshaded/src/main/java/tachyon/client/file/TachyonFSCore.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file;
+
+import java.io.IOException;
+import java.util.List;
+
+import tachyon.TachyonURI;
+import tachyon.thrift.FileInfo;
+
+/**
+ * User facing interface for the Tachyon File System client APIs. File refers to any type of inode,
+ * including folders. Clients should provide their own interface for reading/writing files.
+ */
+interface TachyonFSCore {
```

Comment:
Disregarding naming for a second, when the client gets a `TachyonFileSystem` should they get the interface or the actual implementation? I think the actual implementation should be used since we will offer some additional convenience methods in addition to the bare interface. In this case, I think its weird to return something suffixed with `Impl` since usually those types of classes are hidden behind the interface type.